controller/users.go

```diff
@@ -93,8 +93,117 @@ func (c *UsersController) UpdateUserAsServiceAccount(ctx *app.UpdateUserAsServic
 	}
 	if !isSvcAccount {
 		log.Error(ctx, map[string]interface{}{
-			"err": err,
-		}, "")
+			"identity_id": ctx.ID,
+		}, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to create a user.")))
+	}
+
+	return c.createUserInDB(ctx)
+}
+
+func (c *UsersController) createUserInDB(ctx *app.CreateUserAsServiceAccountUsersContext) error {
+
+	userID, err := uuid.FromString(ctx.Payload.Data.Attributes.UserID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("invalid user id")))
+	}
+
+	id, err := uuid.FromString(ctx.ID)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrBadRequest(errs.New("incorrect identity id")))
+	}
+
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
+
+		var user *account.User
+		var identity *account.Identity
+
+		// Mandatory attributes
+
+		user = &account.User{
+			ID:    userID,
+			Email: ctx.Payload.Data.Attributes.Email,
+		}
+		identity = &account.Identity{
+			ID:       id,
+			Username: ctx.Payload.Data.Attributes.Username,
```

Comment:
We must set ProviderType too. Pass it in the payload from WIT.

Reply:
Thanks. Fixed it along with a test : https://github.com/fabric8-services/fabric8-wit/pull/1635/commits/f9b92d5482954b85b04e7eb2d6f8e21968544e09